source/brailleDisplayDrivers/braillePen.py

```diff
@@ -0,0 +1,143 @@
+#brailleDisplayDrivers/braillePen.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2014 Harpo Sp. z o. o. <http://www.harpo.com.pl>
+
+import braille
+import queueHandler
+from logHandler import log
+from ctypes import *
+from ctypes.wintypes import *
+import time
+import config
+import inputCore
+import hwPortUtils
+import brailleInput
+import wx
+import hwIo
+
+BAUD_RATE = 115200
+TIMEOUT = 0.1
+
+BP12_BRAILLE_KEYS = ("dot1", "dot2", "dot3", "dot4", "dot5", "dot6","dot7","dot8" )
+
+def _findPorts():
+	for portInfo in hwPortUtils.listComPorts(onlyAvailable=True):
+		try:
+			btName = portInfo["bluetoothName"]
+		except KeyError:
+			continue
+		if btName.startswith("BraillePen") or btName.startswith("EasyLink"):
+			yield portInfo["port"]
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	name = "braillePen"
+	# Translators: Names of braille displays.
+	description = _("BraillePen Slim")
+	isThreadSafe = True
+
+	@classmethod
+	def check(cls):
+		return True
+
+	def __init__(self):
+		super(BrailleDisplayDriver, self).__init__()
+		self.numCells = 0
+		for port in _findPorts():
+			try:
+				self._dev = hwIo.Serial(port, baudrate=BAUD_RATE, timeout=TIMEOUT, writeTimeout=TIMEOUT, onReceive=self._serOnReceive)
+			except EnvironmentError:
+				log.info("BraillePen open port error")
```

Comment:
I don't think we need this log call. We definitely don't at level info, since that's meant to be mostly user consumable. You could log it at debug, but hwIo can actually log a whole lot of debugging info anyway, so I don't think it's useful.